### PR TITLE
allow users to have local copy of file to not fetch it over and over

### PR DESCRIPTION
**Pull request by doomedraven**
hello first thanks for tool. would be amazing if you could allow to local users use local file also as option not always fetch online

### CODE REVIEW

File: pyattck/pyattck.py [hunk 1]
```diff
@@ -16,8 +16,15 @@ class Attck(object):
         [Attck]: Returns a Attck object that contains all data from the Mitre ATT&CK Framework
     """
 
-    def __init__(self):
-        self.attck = __MITRE_ATTCK_JSON_URL__
+    def __init__(self, local_file=False):
```
Comment (joshswimlane) [inline pyattck/pyattck.py]: I would change this to `local_file_path` and set it to `None` as the default

File: pyattck/pyattck.py [hunk 2]
```diff
@@ -16,8 +16,15 @@ class Attck(object):
...
+        """
+        Arguments:
+            local_file (str) -- Path where json is placed
```
Comment (joshswimlane) [inline pyattck/pyattck.py]: Same here, change the name of the parameter

File: pyattck/pyattck.py [hunk 3]
```diff
@@ -16,8 +16,15 @@ class Attck(object):
...
+        """
+        if local_file and os.path.exists(local_file):
```
Comment (joshswimlane) [inline pyattck/pyattck.py]: Same here.  Also if `local_file_path` does not exist then we need to create it.

File: pyattck/pyattck.py (hunk 1 above)
Comment (joshswimlane) [reply pyattck/pyattck.py]: Then if `local_file_path` has a value then we should create it.  Also, set a new `self.local_file_path` variable here with the value which can be used in other methods within this class.

File: pyattck/pyattck.py [hunk 4]
```diff
@@ -33,7 +40,11 @@ def attck(self, value):
         Returns:
             (dict) -- Returns the requested json file
         """
-        self._attck = requests.get(value).json()
+        if os.path.exists(value):
```
Comment (joshswimlane) [inline pyattck/pyattck.py]: Check if `self.local_file_path` is True.  If so, then return the file contents.

**Comment by doomedraven** [comment]
thank you

**Comment by joshswimlane** [comment]
No problem, thank you!  It should be up on pypi now as well - new version is 1.0.4

**Comment by doomedraven** [comment]
@joshswimlane  hello, do you know how this code doesn't included anymore in repo? did you wiped that?

**Comment by joshswimlane** [comment]
@doomedraven Let me take a look - i'm making a large update currently and hopefully have it pushed out this week.

**Comment by doomedraven** [comment]
thanks

**Comment by doomedraven** [comment]
the version in pip 1.0.4 is with this changes

**Comment by doomedraven** [comment]
@joshswimlane do you want me to PR this again? or this will be excluded?

**Pull request merged**